nx-X11/programs/Xserver/hw/nxagent/NXdixfonts.c

```diff
@@ -999,1393 +622,86 @@ doListFontsAndAliases(ClientPtr client, LFclosurePtr c)
     return TRUE;
 }
 
+
 int
-ListFonts(ClientPtr client, unsigned char *pattern, unsigned length, 
-          unsigned max_names)
+SetDefaultFontPath(char *path)
 {
-    int         i;
-    LFclosurePtr c;
-
-    /* 
-     * The right error to return here would be BadName, however the
-     * specification does not allow for a Name error on this request.
-     * Perhaps a better solution would be to return a nil list, i.e.
-     * a list containing zero fontnames.
-     */
-    if (length > XLFDMAXFONTNAMELEN)
-	return BadAlloc;
+    unsigned char *cp,
+               *pp,
+               *nump,
+               *newpath;
+    int         num = 1,
+                len,
+                err,
+                size = 0,
+                bad;
 
-    if (!(c = (LFclosurePtr) xalloc(sizeof *c)))
-	return BadAlloc;
-    c->fpe_list = (FontPathElementPtr *)
-	xalloc(sizeof(FontPathElementPtr) * num_fpes);
-    if (!c->fpe_list) {
-	xfree(c);
-	return BadAlloc;
-    }
-    c->names = MakeFontNamesRecord(max_names < nxagentMaxFontNames ? max_names : nxagentMaxFontNames);
-    if (!c->names)
-    {
-	xfree(c->fpe_list);
-	xfree(c);
+    /* get enough for string, plus values -- use up commas */
+#ifdef NX_TRANS_SOCKET
+    len = strlen(_NXGetFontPath(path)) + 1;
+#else
+    len = strlen(path) + 1;
+#endif
+    nump = cp = newpath = (unsigned char *) ALLOCATE_LOCAL(len);
+    if (!newpath)
 	return BadAlloc;
+#ifdef NX_TRANS_SOCKET
+    pp = (unsigned char *) _NXGetFontPath(path);
+#else
+    pp = (unsigned char *) path;
+#endif
+    cp++;
+    while (*pp) {
+	if (*pp == ',') {
+	    *nump = (unsigned char) size;
+	    nump = cp++;
+	    pp++;
+	    num++;
+	    size = 0;
+	} else {
+	    *cp++ = *pp++;
+	    size++;
+	}
     }
-    memmove( c->current.pattern, pattern, length);
-    for (i = 0; i < num_fpes; i++) {
-	c->fpe_list[i] = font_path_elements[i];
-	UseFPE(c->fpe_list[i]);
-    }
-    c->client = client;
-    c->num_fpes = num_fpes;
-    c->current.patlen = length;
-    c->current.current_fpe = 0;
-    c->current.max_names = max_names;
-    c->current.list_started = FALSE;
-    c->current.private = 0;
-    c->haveSaved = FALSE;
-    c->slept = FALSE;
-    c->savedName = 0;
-    doListFontsAndAliases(client, c);
-    return Success;
+    *nump = (unsigned char) size;
+
+    err = SetFontPathElements(num, newpath, &bad, TRUE);
+
+    DEALLOCATE_LOCAL(newpath);
+
+    return err;
 }
 
-int
-doListFontsWithInfo(ClientPtr client, LFWIclosurePtr c)
```

Comment:
doListFontsWithInfo in NXdixfonts.c:

> Modified, but not kept

Oh dear, good catch. This one is nasty as only one line of code is change, but the function voluptous. Sigh.